projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -733,11 +734,10 @@ export class IgxTreeGridComponent extends IgxGridBaseDirective implements GridTy
         //  and if we have transactions we should start pending transaction. This allows
         //  us in case of delete action to delete all child rows as single undo action
         const args = {
-            rowID: rowId,
-            primaryKey: rowId,
+            key: rowId,
```

Comment:
Please type args as `IRowDataCancelableEventArgs`. Here, I see `rowId` is not emitted as part of the object, so we can do the same in grid-base.directive deleteRowById